### ISSUE TITLE
[2.4.0] JdkObsolete complains about method calls on "obsolete" classes (regression)

### ISSUE THREAD

**Issue by ejona86** [google/error-prone#1766]
### Description of the problem / feature request:

JdkObsolete complains about each and every method call on an "obsolete" class. This is strongly related to #1646, but makes the problem worse. To workaround this, we have to suppress JdkObsolete on the entire method or class (since some fields have many references scattered through the class).

### Bugs: what's the simplest, easiest way to reproduce this bug?

We have the code:
```java
      @SuppressWarnings("JdkObsolete") // Hashtable is required
      Hashtable<String, String> env = new Hashtable<>();
      env.put("com.sun.jndi.ldap.connect.timeout", "5000");
      env.put("com.sun.jndi.ldap.read.timeout", "5000");
      DirContext dirContext = new InitialDirContext(env);
```

In error-prone 2.3.4 that code did not produce any warning, because the single warning it would have produced was suppressed. In error-prone 2.4.0 it produces two warnings complaining about the `put()`s:

```
grpc-java/core/src/main/java/io/grpc/internal/JndiResourceResolverFactory.java:210: warning: [JdkObsolete] Hashtable performs synchronization this is usually unnecessary; prefer LinkedHashMap.
      env.put("com.sun.jndi.ldap.connect.timeout", "5000");
             ^
    (see https://errorprone.info/bugpattern/JdkObsolete)
grpc-java/core/src/main/java/io/grpc/internal/JndiResourceResolverFactory.java:211: warning: [JdkObsolete] Hashtable performs synchronization this is usually unnecessary; prefer LinkedHashMap.
      env.put("com.sun.jndi.ldap.read.timeout", "5000");
             ^
    (see https://errorprone.info/bugpattern/JdkObsolete)
```

That does not seem to be helpful, and strongly tempts me to disable JdkObsolete entirely. (LinkedList is already a point of contention due to different memory usage behavior; the check overall has not provided any value and has definitely caused useless effort and almost caused multiple bugs.)

### What version of Error Prone are you using?

Upgraded from error-prone 2.3.4 to error-prone 2.4.0.

###  Have you found anything relevant by searching the web?

Issue #1646

**Comment by cushon** [comment]
The check no longer warns on method invocations as of d450119b23e9c7c68fcfbb3927a3250493ac3ad3.

**Comment by ejona86** [comment]
Excellent! I saw that commit but didn't realize it changed the behavior. Thank you!
